client/source/js/modules/analysis/optimization.html

```diff
@@ -1,370 +1,368 @@
 <div class="page">
+  <p ng-show="needData===true" class="error-hint">Please upload Optima spreadsheet first.</p>
```

Comment:
Same comment as above, it's better to check for true/false values with "===" especially in angular templates because angular uses different kind of javascript parser and it's clearer this way. When i paired with @nikgraf on skype he also said it's better to do explicit checks like this instead of just ng-show="needData".